enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult/InteractiveQuestionResult.tsx

```diff
@@ -84,39 +68,58 @@ export const InteractiveQuestionResult = ({
       className={cx(InteractiveQuestionS.Container, className)}
       style={style}
     >
-      <Group className={InteractiveQuestionS.TopBar} position="apart" p="md">
-        <InteractiveQuestion.BackButton />
-        {withTitle && (customTitle ?? <InteractiveQuestion.Title />)}
-        <Group spacing="xs">
-          {withResetButton && <InteractiveQuestion.ResetButton />}
-          <InteractiveQuestion.ChartTypeDropdown />
-          <InteractiveQuestion.FilterDropdown />
-          <InteractiveQuestion.SummarizeDropdown />
-          <InteractiveQuestion.BreakoutDropdown />
-          <InteractiveQuestion.EditorButton
-            isOpen={questionView === "editor"}
-            onClick={() =>
-              setQuestionView(
-                questionView === "editor" ? "visualization" : "editor",
-              )
-            }
-          />
-
+      <Stack className={InteractiveQuestionS.TopBar} spacing="sm" p="md">
+        <Group position="apart" align="flex-end">
+          <Group spacing="xs">
+            <InteractiveQuestion.BackButton />
+            <ResultTitle
+              withResetButton={withResetButton}
+              customTitle={customTitle}
+              withTitle={withTitle}
+            />
+          </Group>
           {isSaveEnabled && !isSaveModalOpen && (
             <InteractiveQuestion.SaveButton onClick={openSaveModal} />
           )}
         </Group>
-      </Group>
 
-      <Box className={InteractiveQuestionS.Main} p="md" w="100%" h="100%">
-        <Box className={InteractiveQuestionS.Content}>
-          <ContentView
-            questionView={questionView}
-            onReturnToVisualization={() => setQuestionView("visualization")}
+        <Group
+          position="apart"
+          p="sm"
+          bg="var(--mb-color-background-disabled)"
+          style={{ borderRadius: "0.5rem" }}
+        >
+          <Group spacing="xs">
+            <InteractiveQuestion.ChartTypeDropdown />
+            <Divider
+              mx="xs"
+              orientation="vertical"
+              // we have to do this for now because Mantine's divider overrides this color no matter what
+              color="var(--mb-color-border) !important"
```

Comment:
This is interesting because we already defined `theme.fn.themeColor("border")` in the Mantine style overrides. Perhaps we need to define something else in `getDividerOverrides`. Can fix later.

https://github.com/metabase/metabase/blob/cbfdbb2fe6ae1f7d2dc004f3cb01ee0b9166ea6c/frontend/src/metabase/ui/components/utils/Divider/Divider.styled.tsx#L3-L25